src/OAuth2.ts

```diff
@@ -134,6 +134,6 @@ export class OAuth2 {
     }
 
     private makeRedirectURL(token: string): string {
-        return path.join(this.redirectPrefix, token, "authorize");
+        return new URL(`${token}/authorize`, this.redirectPrefix).toString();
```

Comment:
The second parameter of `new URL()` seems to only allow the protocol and domain.
If you add a path (e.g. `https://redir_prefix/my-custom-oauth-endpoint`) the output becomes `https://redir_prefix/token/authorize`.

Reply:
Oh, right you are. Well, failing that I guess we will have to use a library :|